be/src/runtime/plan_fragment_executor.cpp

```diff
@@ -323,7 +323,12 @@ Status PlanFragmentExecutor::open_internal() {
     {
         SCOPED_TIMER(profile()->total_time_counter());
         collect_query_statistics();
-        Status status = _sink->close(runtime_state(), _status);
+        Status status;
+        {
+            boost::lock_guard<boost::mutex> l(_status_lock);
```

Comment:
If you have time. I think we would better use `std::lock_guard` and `std:: mutex ` instead of `boost::lock_guard` and `boost::mutex`.

Reply:
This PR compile failed, I will fix it and change this lock too~